Jenkinsfile

```diff
@@ -63,147 +54,79 @@ pipeline {
-                        ).trim()
-
-                        // example of expected value: 2.33.8
-                        DHIS2_BRANCH_VERSION = sh(
-                            returnStdout: true,
-                            script: "grep DHIS2Version $WORKSPACE/$INPUT_FILE_NAME | awk -F '\"' '{print \$4;}' ").trim().toLowerCase()
-
-                        echo "Package version: ${PACKAGE_VERSION}"
-                        currentBuild.description = "${PACKAGE_VERSION}"
-                        echo "DHIS2 version: ${DHIS2_BRANCH_VERSION}"
+                script {
+                    if (PACKAGE_IS_EXPORTED.toBoolean()) {
+                        sh "cp $WORKSPACE/$PACKAGE_FILE ./test/package_orig.json"
+                    } else {
+                        unstash "$PACKAGE_FILE"
+                        sh "cp $PACKAGE_FILE ./test/package_orig.json"
+                    }
 
-                        def length = sh(
-                            returnStdout: true,
-                            script: "echo -n ${DHIS2_BRANCH_VERSION} | wc -c "
-                        ).trim()
+                    DHIS2_BRANCH_VERSION = sh(returnStdout: true, script: "cat $PACKAGE_FILE | jq -r \".package .DHIS2Version\"").trim()
 
-                        if ("${length}".toInteger() > 4) {
-                            echo "DHIS2 version is a patch version."
-                            CHANNEL = ""
-                        }
-                    }
+                    currentBuild.description = sh(returnStdout: true, script: "cat $PACKAGE_FILE | jq -r \".package .name\"").trim()
                 }
             }
         }
 
-        stage('Metadata validation') {
+        stage('Validate metadata') {
             steps {
                 script {
-                    //This call is done also here just in case the previous stage is skipped
+                    // In case the 'Export package' stage was skipped, the utils repo needs to be cloned again
                     if (!fileExists('dhis2-utils')) {
                         dir('dhis2-utils') {
                             git url: "$UTILS_GIT_URL"
                         }
                     }
 
                     catchError {
-                        sh("python3 -u dhis2-utils/tools/dhis2-metadata-package-validator/metadata_package_validator.py -f $WORKSPACE/$INPUT_FILE_NAME")
+                        sh("python3 -u dhis2-utils/tools/dhis2-metadata-package-validator/metadata_package_validator.py -f $WORKSPACE/$PACKAGE_FILE")
                     }
                 }
             }
         }
 
-        stage('Test empty instance') {
+        stage('Test import in empty instance') {
             steps {
                 script {
                     withDockerRegistry([credentialsId: "docker-hub-credentials", url: ""]) {
-                        d2.startCluster( "${DHIS2_BRANCH_VERSION}", "$PORT", "$CHANNEL")
+                        d2.startCluster("$DHIS2_BRANCH_VERSION", "$PORT")
```

Comment:
Why did you remove channel? It should be either dev or stable, based on the version provided in params, if I remember correctly.

Reply:
The channel defaults to [stable](https://github.com/dhis2/jenkins-pipeline-library/blob/master/vars/d2.groovy#L1), but why would we want to use a "dev" image/branch to test? @stratosilva is there a use-case for that?

Reply:
I think one case could be testing the release candidate - they come from "dev" channel. Not sure if that's something that's done often :)

Reply:
Hi. There was a time when there was a bug which was preventing us from generating a package. A quick bug was implemented in a dev branch but there was not an official release for that fix. So I think there might be a use case for this although it is very rare

Reply:
I've added a check for the DHIS2 version the package was extracted from, in case it was from a "dev" version ([with this commit](https://github.com/dhis2/metadata-package-development/pull/39/commits/53e048ae8f0379d8d91eebb84bc264bb8e03ba07)). 👍